lib/Transforms/ElementwiseToAffine/ElementwiseToAffine.cpp

```diff
@@ -130,10 +130,19 @@ struct ElementwiseToAffine
     RewritePatternSet patterns(context);
 
     patterns.add<ConvertAnyElementwiseMappableOpOnRankedTensors>(context);
-    target.markUnknownOpDynamicallyLegal([](Operation *op) {
-      // TODO (#768): to make this pass more widely applicable, it should take
-      // a dialect and/or list of operations to restrict the conversion to
-      return !isElementwiseMappableOpOnRankedTensors(op);
+    target.markUnknownOpDynamicallyLegal([&](Operation *op) {
+      auto isDialectElegible = llvm::is_contained(
+          restrictDialects, op->getDialect()->getNamespace().str());
+      auto isOpElegible =
+          llvm::is_contained(restrictOps, op->getName().getStringRef().str());
+
+      auto is_EWMtoRT = isElementwiseMappableOpOnRankedTensors(op);
+
+      if (restrictDialects.empty() && restrictOps.empty()) {
+        return !is_EWMtoRT;
+      } else {
+        return (isDialectElegible || isOpElegible) ? !is_EWMtoRT : true;
```

Comment:
You were right, there really is no way to avoid the empty %% empty check, sorry about sending you on a wild goose chase there! 
```suggestion
      bool convertAll = convertDialects.empty() && convertOps.empty();
      bool convertDialect = llvm::is_contained(
          convertDialects, op->getDialect()->getNamespace().str());
      bool convertOp =
          llvm::is_contained(convertOps, op->getName().getStringRef().str());

      if (convertAll || convertDialect || convertOp)
        return !isElementwiseMappableOpOnRankedTensors(op);
      return true;
```